vita/2_build_toolchain.sh

```diff
@@ -5,79 +5,72 @@ set -e
-export URL=`curl "https://api.github.com/repos/vitasdk/autobuilds/releases" | grep "browser_download_url" | grep "linux" | head -n 1 | cut -d '"' -f 4`
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+source $SCRIPT_DIR/../shared/import
 
 # Number of CPU
-NBPROC=$(getconf _NPROCESSORS_ONLN)
+nproc=$(nproc)
+
+# Use ccache?
+test_ccache
 
 if [ ! -f .patches-applied ]; then
 	echo "patching libraries"
-	
-	# Fix compilation problems
-	cp -r icu icu-native
-	patch -Np0 < icu.patch
-
-	# disable pixman examples and tests
-	cd pixman-0.34.0
-	perl -pi -e 's/SUBDIRS = pixman demos test/SUBDIRS = pixman/' Makefile.am
+
+	patches_common
+
+	# Fix mpg123
+	pushd $MPG123_DIR
+	patch -Np1 < $SCRIPT_DIR/../shared/extra/mpg123.patch
 	autoreconf -fi
-	cd ..
+	popd
 
-	# Fix mpg123 compilation
-	patch -Np0 < mpg123.patch
+	# Fix libsndfile
+	pushd $LIBSNDFILE_DIR
+	patch -Np1 < $SCRIPT_DIR/../shared/extra/libsndfile.patch
+	autoreconf -fi
+	popd
 
-	# Fix libsndfile compilation
-	patch -Np0 < libsndfile.patch
+	# Support install for CMAKE_SYSTEM_NAME Generic
+	pushd $WILDMIDI_DIR
+	patch -Np1 < $SCRIPT_DIR/../shared/extra/wildmidi-generic-install.patch
+	# Vita compatibility
+	patch -Np1 < $SCRIPT_DIR/wildmidi-vita.patch
+	popd
 
-	# Fix wildmidi linking
-	patch -Np0 < wildmidi.patch
+	# Fix icu build
+	# Custom patch because vita newlib provides pthread
+	cp -rup icu icu-native
+	patch -Np0 < $SCRIPT_DIR/icu59-vita.patch
 
 	touch .patches-applied
 fi
 
-function set_build_flags {
-	export CC="$TARGET_HOST-gcc"
-	export CXX="$TARGET_HOST-g++"
-	export CFLAGS="-I$WORKSPACE/include -g0 -O2"
-	export CPPFLAGS="$CFLAGS"
-	export LDFLAGS="-L$WORKSPACE/lib"
-}
+cd $WORKSPACE
 
-# Default lib installer
-function install_lib {
-	cd $1
-	./configure --host=$TARGET_HOST --prefix=$PLATFORM_PREFIX --disable-shared --enable-static $2
-	make clean
-	make -j1
-	make install
-	cd ..
-}
+echo "Preparing toolchain"
 
-# Install zlib
-function install_lib_zlib() {
-	cd zlib-1.2.11
-	CHOST=$TARGET_HOST ./configure --static --prefix=$PLATFORM_PREFIX
-	make clean
-	make -j$NBPROC
-	make install
-	cd ..
-}
+export VITASDK=$PWD/vitasdk
+export PATH=$PWD/vitasdk/bin:$PATH
 
-function install_lib_wildmidi() {
-	cd wildmidi-wildmidi-0.4.1
-	cmake . -DCMAKE_SYSTEM_NAME=Generic -DCMAKE_BUILD_TYPE=RelWithDebInfo -DWANT_PLAYER=OFF
-	make clean
-	make
-	cp -up include/wildmidi_lib.h $WORKSPACE/include
-	cp -up libWildMidi.a $WORKSPACE/lib
-	cd ..
+export PLATFORM_PREFIX=$WORKSPACE
+export TARGET_HOST=arm-vita-eabi
+export PKG_CONFIG_PATH=$PLATFORM_PREFIX/lib/pkgconfig
+export PKG_CONFIG_LIBDIR=$PKG_CONFIG_PATH
+export MAKEFLAGS="-j${nproc:-2}"
+
+function set_build_flags {
+	if [ "$ENABLE_CCACHE" ]; then
+		export CC="ccache $TARGET_HOST-gcc"
+		export CXX="ccache $TARGET_HOST-g++"
+	else
+		export CC="$TARGET_HOST-gcc"
+		export CXX="$TARGET_HOST-g++"
+	fi
+	export CFLAGS="-I$PLATFORM_PREFIX/include -g0 -O2 -DPSP2"
+	export CPPFLAGS=$CFLAGS
+	export CXXFLAGS=$CFLAGS
```

Comment:
I know that has been in here before, but is really insane.
Only put `-I` and `-D` stuff in `CPPFLAGS`...